plugin/src/main/java/com/h3xstream/findsecbugs/spring/SpringEntityLeakDetector.java

```diff
@@ -0,0 +1,134 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library.
+ */
+package com.h3xstream.findsecbugs.spring;
+
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.Detector;
+import edu.umd.cs.findbugs.Priorities;
+import edu.umd.cs.findbugs.ba.ClassContext;
+import org.apache.bcel.Repository;
+import org.apache.bcel.classfile.AnnotationEntry;
+import org.apache.bcel.classfile.JavaClass;
+import org.apache.bcel.classfile.Method;
+import org.apache.bcel.generic.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Detects Persistent Objects leak and mass updation
+ *
+ * @author Karan Bansal (Github:karanb192)
+ */
+public class SpringEntityLeakDetector implements Detector {
+	private static final String SPRING_ENTITY_LEAK_TYPE = "SPRING_ENTITY_LEAK";
+	private static final List<String> REQUEST_MAPPING_ANNOTATION_TYPES = Arrays.asList(
+			"Lorg/springframework/web/bind/annotation/RequestMapping;",
+			"Lorg/springframework/web/bind/annotation/GetMapping;",
+			"Lorg/springframework/web/bind/annotation/PostMapping;",
+			"Lorg/springframework/web/bind/annotation/PutMapping;",
+			"Lorg/springframework/web/bind/annotation/DeleteMapping;",
+			"Lorg/springframework/web/bind/annotation/PatchMapping;");
+
+	private static final List<String> ENTITY_ANNOTATION_TYPES = Arrays.asList(
+			"Ljavax/persistence/Entity;",
+			"Ljavax/jdo/spi/PersistenceCapable;",
+			"Lorg/springframework/data/mongodb/core/mapping/Document;");
+
+	private BugReporter reporter;
+
+	public SpringEntityLeakDetector(BugReporter bugReporter) {
+		this.reporter = bugReporter;
+	}
+
+	@Override
+	public void visitClassContext(ClassContext classContext) {
+		JavaClass clazz = classContext.getJavaClass();
+
+		if (hasRequestMapping(clazz)) {
+			Method[] methods = clazz.getMethods();
+			for (Method m: methods) {
+				analyzeMethod(m, classContext);
+			}
+		}
+	}
+
+	private boolean hasRequestMapping(JavaClass clazz) {
+		Method[] methods = clazz.getMethods();
+		for (Method m: methods) {
+			AnnotationEntry[] annotations = m.getAnnotationEntries();
+			m.getReturnType();
+
+			for (AnnotationEntry ae: annotations) {
+				if (REQUEST_MAPPING_ANNOTATION_TYPES.contains(ae.getAnnotationType())) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	private List<String> getAnnotationList(JavaClass javaClass) {
+		ArrayList<String> annotations = new ArrayList<>();
+		for (AnnotationEntry annotationEntry: javaClass.getAnnotationEntries()) {
+			annotations.add(annotationEntry.getAnnotationType());
+		}
+		try {
+			for (JavaClass subclass: javaClass.getSuperClasses()) {
+				annotations.addAll(getAnnotationList(subclass));
+			}
+		} catch (Exception e) {
+		}
+
+		return annotations;
+	}
+	private void analyzeMethod(Method m, ClassContext classContext) {
+		JavaClass clazz = classContext.getJavaClass();
```

Comment:
`hasRequestMapping()` could be applied to only scan method with annotation rather than the complete  controller.
🤔
I am not sure .. We can always adjust if it generate to much false positive.
I will do some tests before the release.